internal/handlers/common/update.go

```diff
@@ -236,6 +243,71 @@ func processPopFieldExpression(doc *types.Document, update *types.Document) (boo
 	return changed, nil
 }
 
+// processRenameFieldExpression changes document according to $rename operator.
+// If the document was changed it returns true.
+func processRenameFieldExpression(doc *types.Document, update *types.Document) (bool, error) {
+	renameExpression := update.SortFieldsByKey()
+
+	var changed bool
+
+	skipErrReg := regexp.MustCompile(`^types\.getByPath: types\.Document\.Get: key not found:.+`)
+
+	for _, key := range renameExpression.Keys() {
+		renameRawValue, err := renameExpression.Get(key)
+		if err != nil {
+			// if $rename field does not exist, don't change anything
+			continue
+		}
+
+		if key == "" || renameRawValue == "" {
+			return changed, NewWriteErrorMsg(ErrEmptyName, "An empty update path is not valid.")
+		}
+
+		sourcePath := types.NewPathFromString(key)
+
+		renameValue, ok := renameRawValue.(string)
+		if !ok {
+			return changed, NewWriteErrorMsg(
+				ErrBadValue,
+				fmt.Sprintf("the 'to' field for $rename must be a string: %s: %v", key, renameRawValue),
```

Comment:
```suggestion
				fmt.Sprintf("The 'to' field for $rename must be a string: %s: %v", key, renameRawValue),
```
Let's be the same response as mongodb shell.

```
test> db.rename.insert({_id:"number1", v: 1})
DeprecationWarning: Collection.insert() is deprecated. Use insertOne, insertMany, or bulkWrite.
{ acknowledged: true, insertedIds: { '0': 'number1' } }
test> db.rename.updateMany({_id:"number1"},{$rename:{"v":1}})
MongoServerError: The 'to' field for $rename must be a string: v: 1
```

Reply:
done